bootstrap/kubeadm/api/v1beta1/kubeadmconfig_types.go

```diff
@@ -707,19 +732,24 @@ type Partition struct {
 	// 'mbr': default and setups a MS-DOS partition table
 	// 'gpt': setups a GPT partition table
 	// +optional
+	// +kubebuilder:validation:MaxLength=3
 	TableType *string `json:"tableType,omitempty"`
 }
 
 // Filesystem defines the file systems to be created.
 type Filesystem struct {
 	// device specifies the device name
+	// +kubebuilder:validation:MaxLength=256
 	Device string `json:"device"`
 	// filesystem specifies the file system type.
+	// +kubebuilder:validation:MaxLength=128
 	Filesystem string `json:"filesystem"`
 	// label specifies the file system label to be used. If set to None, no label is used.
+	// +kubebuilder:validation:MaxLength=512
 	Label string `json:"label"`
 	// partition specifies the partition to use. The valid options are: "auto|any", "auto", "any", "none", and <NUM>, where NUM is the actual partition number.
 	// +optional
+	// +kubebuilder:validation:MaxLength=128
```

Comment:
This one should probably be an enum

Reply:
Sure? "<NUM>, where NUM is the actual partition number."

Reply:
Yeah, the num part is trickier, will think on this one... IntOrString perhaps?

Reply:
Yes it would, so we would likely want to do this as part of the move to v1beta2

Reply:
kk. So leaving it as is for now. Follow-up is tracked in #10852 (same limitation as the other case, someone has to pick up the work)